src/briefcase/platforms/linux/appimage.py

```diff
@@ -166,13 +161,11 @@ def build_app(self, app: AppConfig, **kwargs):
 
             self.logger.info("Configuring Linuxdeploy plugins...", prefix=app.app_name)
             # We need to add the location of the linuxdeploy plugins to the PATH.
-            # However, if we arerunning inside Docker, we need to know the
+            # However, if we are running inside Docker, we need to know the
             # environment *inside* the Docker container.
-            if self.use_docker:
-                echo_cmd = ["/bin/bash", "-c", "echo $PATH"]
-                base_path = self.Docker(self, app).check_output(echo_cmd).strip()
-            else:
-                base_path = self.os.environ["PATH"]
+            with self.run_in_build_environment(app=app):
+                echo_cmd = ["/bin/sh", "-c", "echo $PATH"]
+                base_path = self.subprocess.check_output(echo_cmd).strip()
 
```

Comment:
In the case of `use_docker=False`, this is clearly wildly inefficient but it creates a single apparent codepath for the AppImage build. If that tradeoff doesn't seem worth it, though, I can add the check for `use_docker` back in.

Also, I switched `bash` to `sh` since it is practically guaranteed to be present on any distro.